mlir/test/Ion/QuantumToIon.mlir

```diff
@@ -0,0 +1,485 @@
+    //     levels=[
+    //         #ion.level<
+    //             principal=1,
+    //             spin=1.1,
+    //             orbital=2.2,
+    //             nuclear=3.3,
+    //             spin_orbital=4.4,
+    //             spin_orbital_nuclear=5.5,
+    //             spin_orbital_nuclear_magnetization=6.6,
+    //             energy=8.8
+    //         >,
+    //         #ion.level<
+    //             principal=1,
+    //             spin=1.1,
+    //             orbital=2.2,
+    //             nuclear=3.3,
+    //             spin_orbital=4.4,
+    //             spin_orbital_nuclear=5.5,
+    //             spin_orbital_nuclear_magnetization=6.6,
+    //             energy=8.8
+    //         >
+    //     ],
+    //     transitions=[
+    //         #ion.transition<
+    //             level_0 = #ion.level<
+    //                 principal=1,
+    //                 spin=1.1,
+    //                 orbital=2.2,
+    //                 nuclear=3.3,
+    //                 spin_orbital=4.4,
+    //                 spin_orbital_nuclear=5.5,
+    //                 spin_orbital_nuclear_magnetization=6.6,
+    //                 energy=8.8
+    //             >,
+    //             level_1 = #ion.level<
+    //                 principal=1,
+    //                 spin=1.1,
+    //                 orbital=2.2,
+    //                 nuclear=3.3,
+    //                 spin_orbital=4.4,
+    //                 spin_orbital_nuclear=5.5,
+    //                 spin_orbital_nuclear_magnetization=6.6,
+    //                 energy=8.8
+    //             >,
+    //             einstein_a=10.10
+    //         >,
+    //         #ion.transition<
+    //             level_0 = #ion.level<
+    //                 principal=1,
+    //                 spin=1.1,
+    //                 orbital=2.2,
+    //                 nuclear=3.3,
+    //                 spin_orbital=4.4,
+    //                 spin_orbital_nuclear=5.5,
+    //                 spin_orbital_nuclear_magnetization=6.6,
+    //                 energy=8.8
+    //             >,
+    //             level_1 = #ion.level<
+    //                 principal=1,
+    //                 spin=1.1,
+    //                 orbital=2.2,
+    //                 nuclear=3.3,
+    //                 spin_orbital=4.4,
+    //                 spin_orbital_nuclear=5.5,
+    //                 spin_orbital_nuclear_magnetization=6.6,
+    //                 energy=8.8
+    //             >,
+    //             einstein_a=10.10
+    //         >
+    //     ]
+    // }: !ion.ion
+
+    %1 = quantum.alloc( 2) : !quantum.reg
+    %2 = quantum.extract %1[ 0] : !quantum.reg -> !quantum.bit
+    %3 = quantum.extract %1[ 1] : !quantum.reg -> !quantum.bit
+
+
+    // CHECK-DAG: [[rabi1:%.+]] = arith.constant 1.100000e+00 : f64
+    // CHECK-DAG: [[rabi2:%.+]] = arith.constant 1.230000e+00 : f64
+
+    // CHECK: [[rx1out:%.+]] = ion.parallelprotocol({{%.+}}) : !quantum.bit {
```

Comment:
put in lit test SSA variables instead of using wildcard regex

Reply:
(same for all other pp places)

Reply:
98684af8